pkg/ruler/rulespb/compat_test.go

```diff
@@ -0,0 +1,50 @@
+package rulespb
+
+import (
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/rulefmt"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v3"
+)
+
+func TestProto(t *testing.T) {
+	rules := make([]rulefmt.RuleNode, 0)
+
+	alertNode := yaml.Node{}
+	alertNode.SetString("test_rule")
+	exprNode := yaml.Node{}
+	exprNode.SetString("test_expr")
+
+	testRule := rulefmt.RuleNode{
+		Alert:         alertNode,
+		Expr:          exprNode,
+		Labels:        map[string]string{"label1": "value1"},
+		Annotations:   map[string]string{"key1": "value1"},
+		For:           model.Duration(time.Minute * 2),
+		KeepFiringFor: model.Duration(time.Hour),
+	}
+
+	rules = append(rules, testRule)
+
+	rg := rulefmt.RuleGroup{
+		Name:     "group1",
+		Rules:    rules,
+		Interval: model.Duration(time.Minute),
+	}
+	desc := ToProto("test", "namespace", rg)
+
+	assert.Equal(t, len(rules), len(desc.Rules))
+
+	ruleDesc := desc.Rules[0]
+
+	assert.Equal(t, "test_rule", ruleDesc.Alert)
```

Comment:
should we also verify the keep firing for in alerts field?

Reply:
Added an integration test that validates the KeepFiringSince field in alerts